Source/JavaScriptCore/assembler/MacroAssemblerARMv7.h

```diff
@@ -386,6 +386,14 @@ class MacroAssemblerARMv7 : public AbstractMacroAssembler<Assembler> {
         m_assembler.lsl(dest, src, imm.m_value & 0x1f);
     }
 
+    void lshift32(TrustedImm32 imm, RegisterID shiftAmount, RegisterID dest)
+    {
+        // Clamp the shift to the range 0..31
+        m_assembler.ARM_and(dest, shiftAmount, ARMThumbImmediate::makeEncodedImm(0x1f));
+        move(imm, getCachedAddressTempRegisterIDAndInvalidate());
+        m_assembler.lsl(dest, dataTempRegister, dest);
```

Comment:
Hmm, the `lsl` used `dataTempRegister` but the immediately preceding `move` defines `addressTempRegister` instead?

Reply:
That sounds like it could be it! I'll give it a try.